packages/amplify-category-auth/src/commands/auth/update.ts

```diff
@@ -19,45 +19,49 @@ import { getAuthResourceName } from '../../utils/getAuthResourceName';
-        return context;
-      } else if (serviceMeta.service === AmplifySupportedService.COGNITO && serviceMeta.serviceType === 'imported') {
-        printer.error('Updating imported Auth resource is not supported.');
-        return context;
-      } else if (serviceMeta.service === AmplifySupportedService.COGNITO && !FeatureFlags.getBoolean('auth.forceAliasAttributes')) {
-        const authAttributes = stateManager.getResourceParametersJson(undefined, AmplifyCategories.AUTH, authResourceName);
-        if (authAttributes.aliasAttributes && authAttributes.aliasAttributes.length > 0) {
-          const authUpdateWarning = await BannerMessage.getMessage('AMPLIFY_UPDATE_AUTH_ALIAS_ATTRIBUTES_WARNING');
-          if (authUpdateWarning) {
-            printer.warn(authUpdateWarning);
-          }
+    printer.warn('Project does not contain auth resources. Add auth using `amplify add auth`.');
+    return undefined;
+  }
+  const authResources = Object.keys(existingAuth);
+  for (const authResourceName of authResources) {
+    const serviceMeta = existingAuth[authResourceName];
+    if (serviceMeta.service === AmplifySupportedService.COGNITO && serviceMeta.mobileHubMigrated === true) {
+      printer.error('Auth is migrated from Mobile Hub and cannot be updated with Amplify CLI.');
+      return context;
+    }
+    if (serviceMeta.service === AmplifySupportedService.COGNITO && serviceMeta.serviceType === 'imported') {
+      printer.error('Updating imported Auth resource is not supported.');
+      return context;
+    }
+    if (serviceMeta.service === AmplifySupportedService.COGNITO && !FeatureFlags.getBoolean('auth.forceAliasAttributes')) {
+      const authAttributes = stateManager.getResourceParametersJson(undefined, AmplifyCategories.AUTH, authResourceName);
+      if (authAttributes.aliasAttributes && authAttributes.aliasAttributes.length > 0) {
+        const authUpdateWarning = await BannerMessage.getMessage('AMPLIFY_UPDATE_AUTH_ALIAS_ATTRIBUTES_WARNING');
+        if (authUpdateWarning) {
+          printer.warn(authUpdateWarning);
         }
       }
     }
   }
 
   printer.info('Please note that certain attributes may not be overwritten if you choose to use defaults settings.');
-  const dependentResources = Object.keys(meta).some(e => {
-    return ['analytics', 'api', 'storage', 'function'].includes(e) && Object.keys(meta[e]).length > 0;
-  });
+  const dependentResources = Object.keys(meta).some(e => ['analytics', 'api', 'storage', 'function'].includes(e) && Object.keys(meta[e]).length > 0);
   if (dependentResources) {
     printer.info(messages.dependenciesExists);
   }
   const resourceName = await getAuthResourceName(context);
   await checkAuthResourceMigration(context, resourceName, true);
   const providerPlugin = context.amplify.getPluginInstance(context, servicesMetadata.Cognito.provider);
+  await ensureEnvParamManager();
   context.updatingAuth = providerPlugin.loadResourceParameters(context, 'auth', resourceName);
```

Comment:
nit: Same for this , Directly fetch this from EnironmentManager

Reply:
This method is doing more than just loading from the env param manager. It merges from parameters.json, environment parameters and deployment secrets